### PR TITLE
kubernetes: add integration test for inotify workaround support

### PR DESCRIPTION
**Pull request by egernst**
Add test to ensure fsnotify works and that symlinks are preserved.

Fixes: #4180

Signed-off-by: Eric Ernst <eric_ernst@apple.com>

### CODE REVIEW

**Comment by egernst** [comment]
## Expect this test to fail as is -- will add depends on next...

File: integration/kubernetes/k8s-inotify.bats [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+load "${BATS_TEST_DIRNAME}/tests_common.sh"
+
+setup() {
+	[ "${KATA_HYPERVISOR}" == "firecracker" ] && skip "test not working see: ${fc_limitations}"
+	export KUBECONFIG="${KUBECONFIG:-$HOME/.kube/config}"
+	get_pod_config_dir
+}
+
+@test "configmap update works, and preserves symlinks" {
+	[ "${KATA_HYPERVISOR}" == "firecracker" ] && skip "test not working see: ${fc_limitations}"
```
Comment (wainersm) [inline integration/kubernetes/k8s-inotify.bats:18]: Hi @egernst ,

Always bothered me see the duplication of skip instructions on each tests. In my mind if `setup` skips then the tests should automatically be skipped as well. Then I gave it a try:
```
[wmoschet@wainer-laptop tmp]$ cat foo.bats 
setup() {
	skip "skip setup"
	echo "run setup"
}

@test "test1" {
	skip "skip test1"
}

@test "test2" {
	echo "run test2"
}
[wmoschet@wainer-laptop tmp]$ bats foo.bats 
 - test1 (skipped: skip setup)
 - test2 (skipped: skip setup)

2 tests, 0 failures, 2 skipped

[wmoschet@wainer-laptop tmp]$ bats --version
Bats 1.3.0
```
At least with bats 1.3.0 having `skip` on `setup()` is enough. However I don't know about older versions. We don't set a bats version on any `versions.yaml` (and `tests/.ci/install_bats.sh` pull and install the latest code).

Then I thought "what about the teardown()?". I gave it a try again:
```
[wmoschet@wainer-laptop tmp]$ cat foo2.bats 
setup() {
	skip "skip setup"
	echo "run setup"
}

@test "test1" {
	skip "skip test1"
}

@test "test2" {
	echo "run test2"
}

teardown() {
	/bin/false
}
[wmoschet@wainer-laptop tmp]$ bats foo2.bats 
 ✗ test1
   (from function `teardown' in test file foo2.bats, line 15)
     `/bin/false' failed
 ✗ test2
   (from function `teardown' in test file foo2.bats, line 15)
     `/bin/false' failed

2 tests, 2 failures

```
Yeah, it needs to skip `teardown()` too:
```
[wmoschet@wainer-laptop tmp]$ cat foo3.bats
setup() {
	skip "skip setup"
	echo "run setup"
}

@test "test1" {
	skip "skip test1"
}

@test "test2" {
	echo "run test2"
}

teardown() {
	skip "skip teardown"
}
[wmoschet@wainer-laptop tmp]$ bats foo3.bats 
 - test1 (skipped: skip setup)
 - test2 (skipped: skip setup)

2 tests, 0 failures, 2 skipped
```

Comment (egernst) [reply integration/kubernetes/k8s-inotify.bats:18]: Nice -- good catch!  I'll update the PR to (1) include more tests, (2) skip the in-test skipping!

**Comment by egernst** [comment]
Updated to inspect the symlink handling closer, as well as remove the "skip" from the test case.

**Comment by egernst** [comment]
I see that I need to add a s390x and aarch64 build of our fsnotifu container image.

**Review by mcastelino — APPROVED** [review]

**Review by mxpv — APPROVED** [review]

**Pull request merged**